pkg/invoice/client.go

```diff
@@ -0,0 +1,70 @@
+package invoice
+
+import (
+	"context"
+	"fmt"
+	"github.com/mercadopago/sdk-go/pkg/config"
+	"github.com/mercadopago/sdk-go/pkg/internal/baseclient"
+	"net/url"
+)
+
+const (
+	urlBase   = "https://api.mercadopago.com/authorized_payments"
+	urlSearch = urlBase + "/search"
+	urlWithID = urlBase + "/:id"
+)
+
+// Client contains the methods to interact with the Invoice API.
+type Client interface {
+	// Get finds an invoice by ID.
+	// It is a get request to the endpoint: https://api.mercadopago.com/authorized_payments/{id}
+	// Reference: https://www.mercadopago.com/developers/en/reference/subscriptions/_authorized_payments_id/get
+	Get(ctx context.Context, id string) (*Response, error)
+
+	// Search the invoices for a subscriptions by different parameters.
+	// It is a get request to the endpoint: https://api.mercadopago.com/authorized_payments/search
+	// Reference: https://www.mercadopago.com/developers/en/reference/subscriptions/_authorized_payments_search/get
+	Search(ctx context.Context, request SearchRequest) (*SearchResponsePage, error)
+}
+
+// client is the implementation of Client.
+type client struct {
+	cfg *config.Config
+}
+
+// NewClient returns a new Invoice API Client.
+func NewClient(c *config.Config) Client {
+	return &client{
+		cfg: c,
+	}
+}
+
+func (c *client) Get(ctx context.Context, id string) (*Response, error) {
+	params := map[string]string{
+		"id": id,
+	}
+
+	res, err := baseclient.Get[*Response](ctx, c.cfg, urlWithID, baseclient.WithPathParams(params))
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (c *client) Search(ctx context.Context, request SearchRequest) (*SearchResponsePage, error) {
+	params := request.Parameters()
+
+	parseUrl, err := url.Parse(urlSearch)
```

Comment:
para siglas (url, http, grpc, id, etc) usamos sempre tudo em upper case ou tudo em lower case

invés de
1. baseHttp
2. parsedUrl
3. Id

prefira
1. baseHTTP
2. parsedURL
3. ID ou id
```suggestion
	parsedURL, err := url.Parse(urlSearch)
```